subtests/docker_cli/systemd/systemd.py

```diff
@@ -65,22 +58,32 @@ def sysd_action(action=None, sysd_unit=None):
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def write_unitfile(self, var_to_edit):
+        with open(self.sub_stuff['unit_src'], "r") as srcfile:
+            lines = srcfile.readlines()
+        with open(self.sub_stuff['unit_dest'], "w") as tmpfile:
+            for line in lines:
+                for unitf_var_name, changeto in var_to_edit.items():
+                    if unitf_var_name in line:
+                        str_to_change = \
+                            line.partition(unitf_var_name)[2].strip()
+                        line = line.replace(str_to_change, changeto)
```

Comment:
This is getting a bit too deeply nested, take a step back, consider a simpler way.  What strikes me right away is that the code is modifying the input values, why not just start out with the proper value?

Think, what does python have built-in that is able to take a string, and swap out specific character sequences with keywords, by values found by looking up that keyword in a dictionary?

Oh wait, look, you're already passing in a dictionary to the function anyway :grinning: 

Once you have figured this out, just set up your input (unit) file lines, so they include the tokens ("special" character sequences).  Then setup the values in your dictionary so the output is what you need.

Hint: You're already using something similar in ``sysd_action()``, just w/o the keyword part.